Makefile

```diff
@@ -236,6 +242,11 @@ controller-gen: $(CONTROLLER_GEN) ## Download controller-gen locally if necessar
 $(CONTROLLER_GEN): $(LOCALBIN)
 	test -s $(CONTROLLER_GEN) || GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-tools/cmd/controller-gen@$(CONTROLLER_GEN_VERSION)
 
+.PHONY: yq
+yq: $(YQ) ## Download controller-gen locally if necessary.
+$(YQ): $(LOCALBIN)
```

Comment:
Just a note,  typically for directory existence it's better to use order-only-prerequisites (with |) to avoid triggering the dependency when files are changed, but for our case it's not so important and it's the same style as for other similar targets, so better probably keep it as is.

Reply:
I am not a makefile pro by any means - can you point me to the example of how this should be done properly? I might spawn another PR with the fixes if I get bored or ran out of important tasks during the day ;)

Reply:
Just like
```
$(YQ): | $(LOCALBIN)
``` 
Yes, it deserves a separate change once for all the occasions.

And for the cases where we require binaries in $(LOCALBIN) without other locations checks, for `make` actually it's more natural to have direct file dependency instead of .PHONY ones.

I mean
```
fmt: $(YQ)
    <fmt recipe calling $(YQ)>

$(YQ):  | $(LOCALBIN)
   <build $(YQ)>
```
Then you do not need additional targets and the existence check, it is make's work.
The possible drawbacks:
-  it does not check if execution permission is set, but should not be a problem since in normal scenario it's created by the recipe;
- removing the phony targets causes that if you want just install the binary, you have to type `make $PWD/bin/yq`. 
So, if the targets are widely used better to keep them.